File: problem_1097A/p1097A.cpp
#include<iostream>
#include<string>
using namespace std;

bool compare(string a, string b)
{
    if(a[0]==b[0] || a[1]==b[1])
    return true;
    return false;
}

int main()
{
    string tablecard="",handcard="";

    cin>>tablecard;
    cin.ignore();
    getline(cin,handcard);

    handcard=handcard+" ";

    string w="";
    int i;
    for(i=0;i<handcard.length();i++)
    {
        if(handcard[i]==' ')
        {
            if(compare(tablecard,w))
            {
                cout<<"YES"<<endl;
                return 0;
            }
            w="";
        }
        else
        {
            w=w+handcard[i];
        }
    }
    cout<<"NO"<<endl;
    return 0;
}

File: problem_1097A/p1097A.java
import java.util.*;

public class p1097A
{



public static boolean compare(String a,String b)
{
    if(b.indexOf(a.charAt(0))>-1 || b.indexOf(a.charAt(1))>-1)
    return true;
    return false;
}

public static void main(String[] args)
{
    String tablecard="",handcard="";
    Scanner sc = new Scanner(System.in);

    tablecard=sc.next()+sc.nextLine();
    handcard=sc.next()+sc.nextLine();

    StringTokenizer st = new StringTokenizer(handcard);
    while(st.hasMoreTokens())
    {
        if(compare(tablecard,st.nextToken()))
        {
            System.out.println("YES");
            return;
        }
    }
    System.out.println("NO");

}
}